src/IO/ADIOS/ADIOS2File.cpp

```diff
@@ -109,7 +109,9 @@ void WriteDataset::call(ADIOS2File &ba, detail::BufferedPut &bp)
                     std::nullopt,
                     ba.variables());
 
-                engine.Put(var, ptr);
+                auto do_defer =
+                    ba.m_is_bp5 ? adios2::Mode::Sync : adios2::Mode::Deferred;
+                engine.Put(var, ptr, do_defer);
```

Comment:
This incurs an overhead in EndStep() / PerformDataWrite(). Use Async in those cases.